### PR TITLE
Created test script for case if PreloadedPT without one required parameter

### PR DESCRIPTION
**Pull request by aderiabin**
[Policies]: PreloadedPolicyTable: Validation rules for required parameters

@OVikhrov please review my pull request

### CODE REVIEW

**Comment by aderiabin** [comment]
ATF exits when SDL shuts down. So current script will be failed although it is expected SDL behavior.
ATF improvement needed

**Comment by aderiabin** [comment]
@maria-mih please review my pull request

File: test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+-- Description:
+-- Behavior of SDL during start SDL in case when PreloadedPT has no required parameters
+-- 1. Used preconditions:
+-- Delete files and policy table from previous ignition cycle if any
+-- Do not start default SDL
+-- 2. Performed steps:
+-- Create PreloadedPolicyTable file without one required parameter
+-- Start SDL
+
+-- Requirement summary:
```
Comment (maria-mih) [inline test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua]: Please move this section to top. In last approved version of template we start with it https://github.com/CustomSDL/sdl_atf_test_scripts/pull/32

File: test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+-- [Policies]: PreloadedPolicyTable: Validation rules for required parameters
+--
+-- Expected result:
+-- SDL shuted down
```
Comment (maria-mih) [inline test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua]: Small typo, it should be `shuts` down or is `shut` down

File: test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+config.defaultProtocolVersion = 2
+local preloaded_pt_file_name = "sdl_preloaded_pt.json"
+
+--[[ Required Shared libraries ]]
+local commonFunctions = require ('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+local json = require("modules/json")
+local SDL = require('modules/SDL')
+
+--[[ Preconditions ]]
```
Comment (maria-mih) [inline test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua]: Please add below line:
commonFunctions:newTestCasesGroup ("Preconditions")

File: test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+function Test.backup_preloaded_pt()
```
Comment (maria-mih) [inline test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua]: Please rewrite name of functions in preconditions to be like: 
function Test:Precondition_BackUp_Preloaded_PT

File: test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  os.execute("cp " .. config.pathToSDL .. preloaded_pt_file_name .. ' ' .. config.pathToSDL .. "backup_" .. preloaded_pt_file_name)
+end
+
+function Test.restore_preloaded_pt()
```
Comment (maria-mih) [inline test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua]: Please check comment above about name of functions in section Preconditions.

File: test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+  os.execute("mv " .. config.pathToSDL .. "backup_" .. preloaded_pt_file_name .. " " .. config.pathToSDL .. preloaded_pt_file_name)
+end
+
+function Test.corrupt_preloaded_pt()
```
Comment (maria-mih) [inline test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua]: Please check comment above about name of functions in section Preconditions.

File: test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+    end
+  end
+
+  local dataToWrite = json.encode(data)
+  file = io.open(pathToFile, "w")
+  file:write(dataToWrite)
+  file:close()
+end
+
+function Test.check_sdl()
```
Comment (maria-mih) [inline test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua]: Please check comment above about name of functions in section Preconditions.

File: test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+  EXPECT_HMINOTIFICATION("BasicCommunication.OnSDLClose",{}):Times(1)
+  local status = SDL:CheckStatusSDL()
+  if status == SDL.RUNNING then
+    commonFunctions:userPrint(31, "Test failed: SDL running without required parameters in preloaded policy table")
+    return false
+  end
+  return true
+end
+
+function Test:Precondition_stop_sdl()
```
Comment (maria-mih) [inline test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua]: Please check comment above about name of functions in section Preconditions.

File: test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+  StopSDL(self)
+end
+
+function Test:Precondition()
```
Comment (maria-mih) [inline test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua]: Please check comment above about name of functions in section Preconditions.
Also please make name more descriptive in this case

File: test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+  commonSteps:DeletePolicyTable()
+  self.backup_preloaded_pt()
+  self.corrupt_preloaded_pt()
+end
+
+--[[ Test ]]
```
Comment (maria-mih) [inline test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua]: Please add below line: 
commonFunctions:newTestCasesGroup ("Test")

File: test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+
+function Test:Test_start_sdl()
```
Comment (maria-mih) [inline test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua]: Please change name of functions in this section in a way that they contain TestStep and also make it a more precise e.g. 

function Test:TestStep_Start_Corrupted_SDL

File: test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+  StartSDL(config.pathToSDL, true, self)
+end
+
+function Test:Test()
```
Comment (maria-mih) [inline test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua]: Please check comment above about name of function

File: test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+  self.check_sdl()
+end
+
+--[[ Postconditions ]]
```
Comment (maria-mih) [inline test_scripts/Polices/Validation_of_PolicyTables/preloaded_pt_validation_rules_for_required_parameters.lua]: Please check how this section is organized in https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/98
We again have name of group printed and need to add PostCondition to name of called function.

**Comment by aderiabin** [comment]
@istoimenova please review my pull request

**Comment by istoimenova** [comment]
@LuxoftAKutsan please merge.

**Pull request merged**